official/vision/beta/projects/yolo/dataloaders/yolo_input.py

```diff
@@ -0,0 +1,511 @@
+        comparing the width and height, this is used for the scaled models.  
+      scale_xy: dictionary `float` values inidcating how far each pixel can see 
+        outside of its containment of 1.0. a value of 1.2 indicates there is a 
+        20% extended radius around each pixel that this specific pixel can 
+        predict values for a center at. the center can range from 0 - value/2 
+        to 1 + value/2, this value is set in the yolo filter, and resused here. 
+        there should be one value for scale_xy for each level from min_level to 
+        max_level.
+      best_match_only: `boolean` indicating how boxes are selected for 
+        optimization.
+      coco91to80: `bool` for wether to convert coco91 to coco80 to minimize 
+        model parameters.      
+      darknet: `boolean` indicating which data pipeline to use. Setting to True 
+        swaps the pipeline to output images realtive to Yolov4 and older. 
+      use_tie_breaker: `boolean` indicating whether to use the anchor threshold 
+        value.
+      dtype: `str` indicating the output datatype of the datapipeline selecting 
+        from {"float32", "float16", "bfloat16"}.
+      seed: `int` the seed for random number generation. 
+    """
+    for key in masks.keys():
+      # Assert that the width and height is viable
+      assert output_size[1] % strides[str(key)] == 0
+      assert output_size[0] % strides[str(key)] == 0
+
+    # scale of each FPN level
+    self._strides = strides
+
+    # Set the width and height properly and base init:
+    self._coco91to80 = coco91to80
+    self._image_w = output_size[1]
+    self._image_h = output_size[0]
+
+    # Set the anchor boxes and masks for each scale
+    self._anchors = anchors
+    self._anchor_free_limits = anchor_free_limits
+    self._masks = {
+        key: tf.convert_to_tensor(value) for key, value in masks.items()
+    }
+    self._use_tie_breaker = use_tie_breaker
+    self._best_match_only = best_match_only
+    self._max_num_instances = max_num_instances
+
+    # Image scaling params
+    self._jitter = 0.0 if jitter is None else jitter
+    self._aug_scale_min = aug_scale_min
+    self._aug_scale_max = aug_scale_max
+    self._aug_rand_translate = aug_rand_translate
+    self._aug_rand_perspective = aug_rand_perspective
+
+    # Image spatial distortion
+    self._random_flip = random_flip
+    self._letter_box = letter_box
+    self._random_pad = random_pad
+    self._aug_rand_angle = aug_rand_angle
+
+    # Color space distortion of the image
+    self._aug_rand_saturation = aug_rand_saturation
+    self._aug_rand_brightness = aug_rand_brightness
+    self._aug_rand_hue = aug_rand_hue
+
+    # Set the per level values needed for operation
+    self._scale_xy = scale_xy
+    self._anchor_t = anchor_t
+    self._darknet = darknet
+    self._area_thresh = area_thresh
+
+    keys = list(self._masks.keys())
+
+    if self._anchor_free_limits is not None:
+      maxim = 2000
```

Comment:
Why 2000?

Reply:
addressed

Reply:
ran tests and found anchor free this was the best maximum to prevent clipping.